controlpanel/frontend/mixins.py

```diff
@@ -20,3 +20,19 @@ def form_valid(self, form):
     def get_success_url(self):
         messages.success(self.request, "Successfully updated bedrock status")
         return super().get_success_url()
+
+
+class TextractAccessMixin(OIDCLoginRequiredMixin, PermissionRequiredMixin):
```

Comment:
I am wondering if it is worth having this as a mixin if it is only used in one place? The bedrock one gets used in two places (apps or users), is there any likelihood of textract being enabled for users? If not, I'd probably just keep all the logic on the actual view class.

Or alternatively, since this essentially does the same thing as the `BedrockAccessMixin` you could refactor to combine the two to a more general mixin that allows a policy to be attached, like a `AttachPolicyMixin` or something like that? You'd need to set the `fields`, `success_message` on the view, and dynamically get the method that gets called on the object. Could then be reused if we need to add any more policies that can be attached.

Reply:
I've updated the mixin now. See what you think of the implementation